crates/shared/src/trade_finding/external.rs

```diff
@@ -186,17 +205,41 @@ mod dto {
         pub deadline: chrono::DateTime<chrono::Utc>,
     }
 
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize)]
+    pub enum QuoteKind {
+        Legacy(LegacyQuote),
+        #[allow(unused)]
+        Regular(Quote),
+    }
+
     #[serde_as]
     #[derive(Clone, Debug, Deserialize)]
     #[serde(rename_all = "camelCase")]
-    pub struct Quote {
+    pub struct LegacyQuote {
         #[serde_as(as = "HexOrDecimalU256")]
         pub amount: U256,
         pub interactions: Vec<Interaction>,
         pub solver: H160,
         pub gas: Option<u64>,
+        pub tx_origin: Option<H160>,
+    }
+
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    #[allow(unused)]
+    pub struct Quote {
+        #[serde_as(as = "HashMap<_, HexOrDecimalU256>")]
+        pub clearing_prices: HashMap<H160, U256>,
         #[serde(default)]
+        pub pre_interactions: Vec<Interaction>,
+        pub interactions: Vec<Interaction>,
```

Comment:
`interactions` could also use `#[serde(default)]` since it's conceptually possible to quote against an order without need to use any regular AMM interactions.